app/src/org/commcare/network/OkHttpBuilderCustomConfig.java

```diff
@@ -0,0 +1,35 @@
+package org.commcare.network;
+
+import android.os.Build;
+
+import org.commcare.core.network.HttpBuilderConfig;
+
+import okhttp3.OkHttpClient;
+
+/**
+ * The purpose of this class is to offer a single entry point for additional OkHttpClient.Builder
+ * configs. As it stands, it serves to attach the ISRG Root X1 Certificate and enable/disable
+ * Certificate Transparency
+ */
+public class OkHttpBuilderCustomConfig implements HttpBuilderConfig {
+    private CTInterceptorConfig ctInterceptorConfig;
+    private ISRGCertConfig isrgCertConfig;
+
+    public OkHttpBuilderCustomConfig(){
+         ctInterceptorConfig = new CTInterceptorConfig();
+         isrgCertConfig = new ISRGCertConfig();
```

Comment:
Lets initialise them in ` performCustomConfig` itself. That way we can make sure they are only initialised if needed. For ex. we don't need to init  `ISRGCertConfig` if ` if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N_MR1) {` is not true.

Reply:
good point

Reply:
can you say more to what you are thinking ?

Reply:
Considering also that all the `CommCareNetworkServiceGenerator` content is static we don't actually need instances of `CTInterceptorConfig` and `ISRGCertConfig` to apply these custom configurations, right?

Reply:
@shubham1g5 in concrete terms, [this](https://github.com/dimagi/commcare-android/pull/2681/commits/6a720db57319cf69d38f48e857bf5ac9693198a5) is what I'm thinking.